src/main/java/org/mifos/connector/mpesa/camel/routes/PaybillRoute.java

```diff
@@ -27,80 +29,82 @@
+                    logger.debug("Paybill Response Body : {}", paybillResponseBodyString);
+                    logger.debug("Reconciled : {}", paybillResponse.getBoolean("reconciled"));
+                    GsmaTransfer gsmaTransfer = mpesaUtils.createGsmaTransferDTO(paybillResponse);
+                    e.getIn().setHeader("accountHoldingInstitutionId", paybillResponse.getString("accountHoldingInstitutionId"));
+                    e.getIn().setHeader("amsName", paybillResponse.getString("amsName"));
+                    e.getIn().setHeader("reconciled", paybillResponse.getBoolean("reconciled"));
+                    e.getIn().setHeader("mpesaTxnId", paybillResponse.getString("transactionId"));
+                    e.setProperty("channelUrl", channelUrl);
+                    return gsmaTransfer.toString();
+                })
+                .toD("${header.channelUrl}" + "/channel/gsma/transaction}" + "?bridgeEndpoint=true&throwExceptionOnFailure=false")
                 .process(e -> {
-                    String paybillRequestBodyString = e.getIn().getBody(String.class);
-                    JSONObject paybillRequest = new JSONObject(paybillRequestBodyString);
-                    logger.debug("Paybill Request Body : {}", paybillRequestBodyString);
-                    logger.debug("Reconciled : {}", paybillRequest.getBoolean("reconciled"));
-                    e.setProperty("MPESA_VALIDATION_WEBHOOK_SUCCESS", paybillRequest.getBoolean("reconciled"));
-
-                    Map<String, Object> variables = new HashMap<>();
-                    variables.put("timer", timer);
-                    variables.put("paybillRequestBody", paybillRequestBodyString);
-                    variables.put("partyLookupFailed", !(paybillRequest.getBoolean("reconciled")));
-                    variables.put("confirmationReceived", false);
-                    //Starting the paybill workflow
-                    String mpesaTransactionId = paybillRequest.getString("transaction_id");
-                    String dfspId = e.getIn().getHeader("dfspId").toString();
-                    String transactionId = zeebeProcessStarter.startZeebeWorkflowPaybill("mpesa_paybill-" + dfspId, variables);
-                    // Storing the external txn id and workflow txn id
-                    logger.debug("MPESA Txn ID :{}", mpesaTransactionId);
-                    logger.debug("Txn ID :{}", transactionId);
-                    hm.put(mpesaTransactionId, transactionId);
-                    // Sending mpesa specific response
+                    // Setting mpesa specifc response
+                    String channelResponseBodyString = e.getIn().getBody(String.class);
+                    JSONObject channelResponse = new JSONObject(channelResponseBodyString);
+                    String mpesaTxnId = e.getIn().getHeader("mpesaTxnId").toString();
+                    Boolean reconciled = Boolean.valueOf(e.getIn().getHeader("reconciled").toString());
```

Comment:
What is this header? Is this part of the design?

Reply:
It is used to store it in redis and later resume the workflow